src/tests/performance/prio_test.rs

```diff
@@ -0,0 +1,330 @@
+    //
+    // Test should be inspected manually to check how other malicious nodes affect the latency of
+    // the node under test.
+    //
+    // Sample results:
+    // ┌──────────────────┬───────────────────┬────────────┬────────────┬────────────┬────────────────┬────────────────┬────────────┐
+    // │  low-prio peers  │  high-prio peers  │  requests  │  min (ms)  │  max (ms)  │  std dev (ms)  │  completion %  │  time (s)  │
+    // ├──────────────────┼───────────────────┼────────────┼────────────┼────────────┼────────────────┼────────────────┼────────────┤
+    // │                1 │                 1 │        300 │          1 │          2 │              1 │         100.00 │       0.46 │
+    // ├──────────────────┼───────────────────┼────────────┼────────────┼────────────┼────────────────┼────────────────┼────────────┤
+    // │                1 │                50 │        300 │          1 │          2 │              1 │         100.00 │       0.48 │
+    // ├──────────────────┼───────────────────┼────────────┼────────────┼────────────┼────────────────┼────────────────┼────────────┤
+    // │                1 │               100 │        300 │          1 │          3 │              1 │         100.00 │       0.53 │
+    // ├──────────────────┼───────────────────┼────────────┼────────────┼────────────┼────────────────┼────────────────┼────────────┤
+    // │                1 │               200 │        300 │          1 │          3 │              1 │         100.00 │       0.55 │
+    // ├──────────────────┼───────────────────┼────────────┼────────────┼────────────┼────────────────┼────────────────┼────────────┤
+    // │                1 │               300 │        300 │          1 │          5 │              1 │         100.00 │       0.75 │
+    // ├──────────────────┼───────────────────┼────────────┼────────────┼────────────┼────────────────┼────────────────┼────────────┤
+    // │                1 │               400 │        300 │          1 │          4 │              1 │         100.00 │       0.76 │
+    // ├──────────────────┼───────────────────┼────────────┼────────────┼────────────┼────────────────┼────────────────┼────────────┤
+    // │                1 │               799 │        300 │          1 │          7 │              1 │         100.00 │       0.78 │
+    // └──────────────────┴───────────────────┴────────────┴────────────┴────────────┴────────────────┴────────────────┴────────────┘
+    // *NOTE* run with `cargo test --release  tests::performance::prio -- --nocapture`
+    // Before running test generate dummy devices with different ips using toos/ips.py
+
+    let h_prio_peers = vec![1, 50, 100, 200, 300, 400, 799];
+    let l_prio_peers = 1;
+
+    let mut table = RequestsTable::default();
+
+    for hp_peer_iter_cnt in h_prio_peers {
+        // synth_count malicious tasks plus one normal synth_node
+        let barrier = Arc::new(Barrier::new(hp_peer_iter_cnt + 1));

```

Comment:
yeah, but now mixing `hp_peer_iter_cnt` with `l_prio_peers` feels odd

```suggestion
    let h_prio_peer_set = vec![1, 50, 100, 200, 300, 400, 799];

    let mut table = RequestsTable::default();
    
    let l_prio_peers = 1;
    for h_prio_peers in h_prio_peer_set {
        let total_peers = l_prio_peers + l_prio_peers;
            
        // synth_count malicious tasks plus one normal synth_node
        let barrier = Arc::new(Barrier::new(total_peers));
```

Reply:
Done.